bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleEngineImpl.java

```diff
@@ -297,51 +279,61 @@ private void scheduleStartupRules() {
         if (startupJob != null && !startupJob.isCancelled() && !startupJob.isDone()) {
             startupJob.cancel(true);
         }
-        startupJob = scheduler.schedule(startupRunnable, 5, TimeUnit.SECONDS);
+        startupJob = scheduler.schedule(() -> {
+            runStartupRules();
+        }, 5, TimeUnit.SECONDS);
     }
 
     private void runStartupRules() {
         if (triggerManager != null) {
             Iterable<Rule> startupRules = triggerManager.getRules(STARTUP);
-            List<Rule> executedRules = Lists.newArrayList();
 
             for (Rule rule : startupRules) {
-                try {
-                    Script script = scriptEngine.newScriptFromXExpression(rule.getScript());
-                    logger.debug("Executing startup rule '{}'", rule.getName());
-                    RuleEvaluationContext context = new RuleEvaluationContext();
-                    context.setGlobalContext(RuleContextHelper.getContext(rule, injector));
-                    script.execute(context);
-                    executedRules.add(rule);
-                } catch (ScriptExecutionException e) {
-                    if (!e.getMessage().contains("cannot be resolved to an item or type")) {
-                        logger.error("Error during the execution of startup rule '{}': {}",
-                                new Object[] { rule.getName(), e.getCause().getMessage() });
-                        executedRules.add(rule);
-                    } else {
-                        logger.debug("Execution of startup rule '{}' has been postponed as items are still missing: {}",
-                                rule.getName(), e.getMessage());
+                scheduler.schedule(() -> {
```

Comment:
Instead of schedule with a zero delay wouldn't it make sense to use `execute` instead (as long as we are not interested in the ScheduledFuture)?

Reply:
Just one question regarding this: The JavaDoc on `execute()` says:

> The command may execute in a new thread, in a pooled thread, or in the calling thread, at the discretion of the Executor implementation."

having it run "in the calling thread" is not exactly what is desired here. How can one be sure it is async then?

Reply:
Damn, I have the `execute` of the `ScheduledThreadPoolExecutor` in mind.
It is the only implementation in the JRE of the `ScheduledExecutorService` and define the execute this way:
"Executes command with zero required delay. This has effect equivalent to `schedule(command, 0, anyUnit)`. ..."

But you are right, we return an implementation of the interface and so it is not defined that execute will be handled async.

Reply:
So would `submit`have been the better option then?

Reply:
I think so.